### DISCUSSION TITLE
Android subscription not getting acknowledged after purchase

### DISCUSSION THREAD

**Discussion by asaf959** [hyochan/react-native-iap#2816] [Q&A]
### Description
When we purchase subscription and call `await acknowledgePurchaseAndroid(token)`, I am getting the error shown in the attached screenshot. I printed the token variable and it is giving me the token but it is not getting acknowledged.

### Expected Behavior
N/A 

### Screenshots
![Screenshot_2024-08-08-15-52-22-06_e05ded82875fa0589f1afb6b8d7aa7d8](https://github.com/user-attachments/assets/774dda04-830c-49c6-8890-e6a7cd5742ca)

### Environment:

react-native-iap: 12.13.1
react-native: 0.72.5
Platforms (iOS, Android, emulator, simulator, device)

**Comment by therealaezel404** [comment]
I'm having the same error but there is a comment on this link [/issues/1006] stating that acknowledgment is not necessary depending on the second parameter, which is whether the item is consumable or not.

However, upon following the statement:

`try {
    purchaseUpdateSubscription = IAP.purchaseUpdatedListener(
        async (purchase = SubscriptionPurchase) => {
        const receipt = purchase.transactionReceipt;
        if (receipt) {
            await IAP.finishTransaction({
                purchase: purchase,
                inConsumable: false
            }).then((result) => {
                console.log("result: ", result)
            }).catch(err => {
                console.error(err.code, err.message);
            });
        }
    })
} catch (error) {
    console.log("current purchase error handle: ", error.message)
}`

It gave me this error:

**E_SERVICE_ERROR The service is unreachable. This may be your internet connection, or the Play Store may be down.**

I hope someone can help. TIA!

Environment:
react-native-iap: 12.15.1
react-native: 0.74.2
Platforms (Android, emulator, simulator, device)

**Comment by hyochan** [comment]
## This API No Longer Exists in v14+

`acknowledgePurchaseAndroid` has been **removed** in react-native-iap v14+.

You are using v12.x which is **no longer supported**.

## Migration to v14+

In v14+, use `finishTransaction` which handles acknowledgment automatically:

```typescript
import { useIAP } from 'react-native-iap';

const { finishTransaction } = useIAP({
  onPurchaseSuccess: async (purchase) => {
    // Validate purchase on your server first, then finish
    await finishTransaction({
      purchase,
      isConsumable: false, // true for consumables
    });
  },
});
```

Or using the root API:

```typescript
import { finishTransaction, purchaseUpdatedListener } from 'react-native-iap';

purchaseUpdatedListener(async (purchase) => {
  if (purchase.transactionReceipt) {
    // finishTransaction handles acknowledgment internally
    await finishTransaction({
      purchase,
      isConsumable: false,
    });
  }
});
```

## About E_SERVICE_ERROR

The "service is unreachable" error typically indicates:

- Network connectivity issues
- Google Play Store service down
- Emulator without Google Play Services
- Testing on device without proper Google account

## Please Upgrade

Versions below v14 are **no longer supported**. Use at your own risk.

- [Migration Guide (v13 to v14)](https://hyochan.github.io/react-native-iap/docs/guides/migration-v13-to-v14)
